docs/content/development/modules.md

```diff
@@ -8,7 +8,7 @@ Druid uses a module system that allows for the addition of extensions at runtime
 
 ## Writing your own extensions
 
-Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of a few things.
+Druid's extensions leverage Guice in order to add things at runtime.  Basically, Guice is a framework for Dependency Injection, but we use it to hold the expected object graph of the Druid process.  Extensions can make any changes they want/need to the object graph via adding Guice bindings.  While the extensions actually give you the capability to change almost anything however you want, in general, we expect people to want to extend one of the things listed below.  This means that we honor semantic versioning for changes that affect the interfaces called out on this page, but other interfaces are deemed "internal" anad can be changed in an incompatible manner even between patch releases.
```

Comment:
@cheddar can you clarify what "honor semantic versioning" means to us? IIRC, the "official" semver thingy says that no guarantees whatsoever are made if the version number starts with "0". I think we want to say something else, like: so long as the version number starts with 0, these interfaces will remain stable within a 0.X line.